File: ios/SuperheroApp/DataSource/CharactersDataSource.swift
//
//  CharactersDataSource.swift
//  SuperheroApp
//
//  Created by Balazs Varga on 2018. 02. 11..
//  Copyright © 2018. W.UP. All rights reserved.
//

import Common

protocol CharactersDataSource {

    func loadCharacters(page: CommonPage, complete: @escaping ([CommonCharacter]) -> Void, fail: @escaping () -> Void)

    func loadCharacter(characterId: Int32, complete: @escaping (CommonCharacter?) -> Void, fail: @escaping () -> Void)

    func saveCharacters(characters: [CommonCharacter], complete: @escaping () -> Void, fail: @escaping () -> Void)
}

File: common/src/main/kotlin/digital/wup/superheroapp/datasource/CharactersDataSource.kt
package digital.wup.superheroapp.datasource

import digital.wup.superheroapp.characters.domain.usecase.model.Page
import digital.wup.superheroapp.characters.domain.usecase.model.Character

interface CharactersDataSource {
    fun loadCharacters(page: Page, complete: (List<Character>) -> Unit, fail: () -> Unit)

    fun loadCharacter(characterId: Int, complete: (Character?) -> Unit, fail:() -> Unit)

    fun saveCharacters(characters: List<Character>, complete: () -> Unit, fail: () -> Unit)
}